### PR TITLE
Feat/SPN-1 add user module

### PR DESCRIPTION
**Pull request by aelshinawy**
Addresses ticket no **[SPN-1](https://syntaxnexus.youtrack.cloud/issue/SPN-1) Create User Module**

### CODE REVIEW

File: backend/build.gradle.kts [hunk 1]
```diff
@@ -28,6 +28,7 @@ dependencies {
 	implementation ("io.vavr:vavr:${rootProject.extra["vavrVersion"]}")
 	implementation ("org.postgresql:postgresql")
 	compileOnly ("org.projectlombok:lombok")
+	implementation("io.vavr:vavr-jackson:0.10.3")
```
Comment (Riolite) [inline backend/build.gradle.kts:31]: Please make this version configurable. You can use vavrVersion and set it to 0.10.3

File: backend/src/main/java/com/syntaxnexus/spendora/user/UserService.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+
+
+    public Try<User> findById(UUID id) {
+        return Try.of(() -> userRepository.findById(id));
+    }
+
+    public Try<List<User>> findAll() {
+        return Try.of(userRepository::findAll);
```
Comment (Riolite) [inline backend/src/main/java/com/syntaxnexus/spendora/user/UserService.java:29]:
```suggestion
        return Try.of(userRepository::findAll).orElse(Collections.emptyList());

```

Comment (Riolite) [reply backend/src/main/java/com/syntaxnexus/spendora/user/UserService.java:29]: You can use orElse here if you want to handle failure with an emptyList.

File: backend/src/main/java/com/syntaxnexus/spendora/user/UserController.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    @GetMapping(produces = "application/json")
+    public ResponseEntity<List<User>> getAllUsers() throws Throwable {
+
+        return userService.findAll().map(ResponseEntity::ok).getOrElseThrow((e) -> e);
+    }
+
+    @GetMapping(value="{userId}", produces = "application/json")
+    public ResponseEntity<User> getUser(@PathVariable UUID userId) {
+
+        return Try.of(()-> userService.findById(userId).get())
```
Comment (Riolite) [inline backend/src/main/java/com/syntaxnexus/spendora/user/UserController.java:36]: No need to call Try.get() and wrap it in a Try here.

**Review by Riolite — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/syntaxnexus/spendora/user/UserService.java (hunk 2 above)
Comment (aelshinawy) [reply backend/src/main/java/com/syntaxnexus/spendora/user/UserService.java:29]: Can't because it violates the function return value

**Review by Riolite — APPROVED** [review]

**Comment by Riolite** [comment]
Good job!!

**Pull request merged**